libqtile/config.py

```diff
@@ -609,34 +609,28 @@ def __init__(self, title=None, wm_class=None, role=None, wm_type=None,
         self._rules += [('net_wm_pid', w) for w in net_wm_pid]
 
     def compare(self, client):
-        for _type, rule in self._rules:
+
+        def _comparator(_type, rule, value):
             if _type == "net_wm_pid":
-                def match_func(value):
-                    return rule == value
-            else:
-                match_func = getattr(rule, 'match', None) or \
-                    getattr(rule, 'count')
-
-            if _type == 'title':
-                value = client.name
-            elif _type == 'wm_class':
-                value = client.window.get_wm_class()
-                _match_func = match_func
-
-                def match_func(values):
-                    return any(_match_func(v) for v in values)
-            elif _type == 'wm_instance_class':
-                value = client.window.get_wm_class()
-                if value:
-                    value = value[0]
-            elif _type == 'wm_type':
-                value = client.window.get_wm_type()
-            elif _type == 'net_wm_pid':
-                value = client.window.get_net_wm_pid()
-            else:
-                value = client.window.get_wm_window_role()
+                return rule == value
 
-            if value and match_func(value):
+            comp = getattr(rule, 'match', None) or getattr(rule, 'count')
+
+            if _type == "wm_class":
+                return any(comp(v) for v in value)
+            return comp(value)
+
+        prop_getters = {
+                'title': lambda c: c.name,
+                'wm_class': lambda c: c.window.get_wm_class(),
+                'wm_instance_class': lambda c: c.window.get_wm_class() and c.window.get_wm_class()[0],
+                'wm_type': lambda c: c.window.get_wm_type(),
+                'net_wm_pid': lambda c: c.window.get_net_wm_pid(),
+                'role': lambda c: c.window.get_wm_window_role(),
+                }
+
+        for _type, rule in self._rules:
+            if _comparator(_type, rule, prop_getters[_type](client)):
```

Comment:
Maybe I should get rid of the "client" parameter here to not add any limitation (just use client from the scope and have no parameter for the lambdas)